hello.py

```diff
@@ -386,180 +399,460 @@ def image_ref(self, repo):
+        cmd += runargs.arg
+        return cmd
+
+    def create_cmd_arg_wait_cmd(self, image_ref, cid, runargs):
+        cmd = 'nerdctl --snapshotter %s create --net=host ' % (self.snapshotter)
+        if len(runargs.env) >0:
+            env = ' '.join(['--env %s=%s' % (k,v) for k,v in runargs.env.items()])
+            cmd = "%s %s " % (cmd, env)
+        for a,b in runargs.mount:
+            a = os.path.join(os.path.dirname(os.path.abspath(__file__)), a)
+            a = tmp_copy(a)
+            cmd += '--volume %s:%s ' % (a,b)
+        cmd += '--name=%s %s' % (cid, image_ref)
+        if len(runargs.arg) > 0:
+            cmd = "%s -- %s " % (cmd, runargs.arg)
+        
+        return cmd
+
+    def create_cmd_stdin_cmd(self, image_ref, cid, runargs):
+        cmd = 'nerdctl --snapshotter %s create --net=host ' % (self.snapshotter)
+        for a,b in runargs.mount:
+            a = os.path.join(os.path.dirname(os.path.abspath(__file__)), a)
+            a = tmp_copy(a)
+            cmd += '--volume %s:%s ' % (a,b)
+        cmd += '--name=%s %s' % (cid, image_ref)
+        if runargs.stdin_sh:
+            cmd = '%s -- %s' % (cmd, runargs.stdin_sh) # e.g., sh -c
+        return cmd
+    
+    def create_cmd_url_wait_cmd(self, image_ref, cid, runargs):
+        cmd = 'nerdctl --snapshotter %s create --net=host ' % (self.snapshotter)
+        for a,b in runargs.mount:
+            a = os.path.join(os.path.dirname(os.path.abspath(__file__)), a)
+            a = tmp_copy(a)
+            cmd += '--volume %s:%s ' % (a,b)
+        if len(runargs.env) >0:
+            env = ' '.join(['--env %s=%s' % (k,v) for k,v in runargs.env.items()])
+            cmd = "%s %s " % (cmd, env)
+        cmd += '--name=%s %s' % (cid, image_ref)
+        if len(runargs.arg) > 0:
+            cmd = "%s -- %s " % (cmd, runargs.arg)
+        return cmd
+
+    def task_start_cmd(self, cid, iteration: bool):
+        if iteration:
+            return 'nerdctl --snapshotter %s start -a %s' % (self.snapshotter, cid)
+        else:
+            return 'nerdctl --snapshotter %s start %s' % (self.snapshotter, cid)
+
+    def task_kill_cmd(self, cid):
+        cmd = 'nerdctl --snapshotter %s stop %s' % (self.snapshotter, cid)
+        return cmd
 
+    def cleanup(self, image_ref, cid):
+        print("Cleaning up environment for %s ..." % cid)
+        cmd = self.task_kill_cmd(cid)
+        print(cmd)
+        rc = os.system(cmd) # sometimes containers already exit. we ignore the failure.
+        cmd = 'nerdctl --snapshotter %s rm -f %s' % (self.snapshotter, cid)
+        print(cmd)
+        rc = os.system(cmd)
+        assert(rc == 0)
+        cmd = md = 'nerdctl --snapshotter %s rmi -f %s' % (self.snapshotter, image_ref)
+        print(cmd)
+        rc = os.system(cmd)
+        assert(rc == 0)
+    
+    def optimize_echo_hello_cmd(self, repo, optimize_bin, accessed_files_dir, target, period):
```

Comment:
Can we move optimizer wrapper code to a new python file ?

Reply:
The arguments for optimization are from `BenchRunner`, which is a class in `hello.py`.